phpBB/phpbb/attachment/upload.php

```diff
@@ -326,10 +333,36 @@ protected function check_attach_quota()
 	/**
 	 * Check if there is enough free space available on disk
 	 *
-	 * @return bool True if disk space is available, false if not
+	 * @return bool True if disk space is available or not limited, false if not
 	 */
 	protected function check_disk_space()
 	{
+		try
+		{
+			$free_space = $this->storage->free_space();
+
+			if ($free_space <= $this->file->get('filesize'))
+			{
+				if ($this->auth->acl_get('a_'))
+				{
+					$this->file_data['error'][] = $this->language->lang('ATTACH_DISK_FULL');
+				}
+				else
+				{
+					$this->file_data['error'][] = $this->language->lang('ATTACH_QUOTA_REACHED');
+				}
+				$this->file_data['post_attach'] = false;
+
+				$this->file->remove($this->storage);
```

Comment:
not sure it should be removed here and not in the calling method

Reply:
Here